### PR TITLE
Cassandra 18787: Replace or remove obsolete functions and simplify cql_verison handling

### PR DESCRIPTION
**Pull request by bschoening**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

File: pylib/cqlshlib/cqlshmain.py [hunk 1]
```diff
@@ -277,22 +276,10 @@ def __repr__(self):
...
 class FormatError(DecodeError):
     verb = 'format'
 
 
-def full_cql_version(ver):
-    while ver.count('.') < 2:
-        ver += '.0'
-    ver_parts = ver.split('-', 1) + ['']
-    vertuple = tuple(list(map(int, ver_parts[0].split('.'))) + [ver_parts[1]])
-    return ver, vertuple
```
Comment (ekaterinadimitrova2) [inline pylib/cqlshlib/cqlshmain.py:293]: The old one was handling alpha beta versions; the new one will raise a ValueError if I read it correctly.

File: pylib/cqlshlib/cqlshmain.py [hunk 2]
```diff
@@ -277,22 +276,10 @@ def __repr__(self):
         return '<%s %s>' % (self.__class__.__name__, self.message())
 
 
-def maybe_ensure_text(val):
```
Comment (ekaterinadimitrova2) [inline pylib/cqlshlib/cqlshmain.py:280]: Why was this check removed?

File: pylib/cqlshlib/cqlshmain.py [hunk 3]
```diff
@@ -2166,15 +2150,12 @@ def read_options(cmdlineargs, environment=os.environ):
         print("\nWarning: Using a password on the command line interface can be insecure."
               "\nRecommendation: use the credentials file to securely provide the password.\n", file=sys.stderr)
 
-    # Make sure some user values read from the command line are in unicode
-    options.execute = maybe_ensure_text(options.execute)
```
Comment (ekaterinadimitrova2) [inline pylib/cqlshlib/cqlshmain.py:2170]: Same questions as before

File: pylib/cqlshlib/cqlshmain.py [hunk 4]
```diff
@@ -2209,22 +2187,16 @@ def read_options(cmdlineargs, environment=os.environ):
         else:
             options.color = should_use_color()
 
-    if options.cqlversion is not None:
-        options.cqlversion, cqlvertup = full_cql_version(options.cqlversion)
-        if cqlvertup[0] < 3:
-            parser.error('%r is not a supported CQL version.' % options.cqlversion)
```
Comment (ekaterinadimitrova2) [inline pylib/cqlshlib/cqlshmain.py:2215]: Why is this removed?

**Review by ekaterinadimitrova2 — COMMENTED** [review]
Left a few comments/questions.
Also, what is the story behind field_size_limit? Why it was needed and was it a leftover or is it a bug that it is not used anymore?

File: pylib/cqlshlib/cqlshmain.py (hunk 2 above)
Comment (bschoening) [reply pylib/cqlshlib/cqlshmain.py:280]: Since Python 3.0, all strings are Unicode.  It appears left over from python 2.x.  Its used only on four command line args to "make sure user values .. are in unicode"

File: pylib/cqlshlib/cqlshmain.py (hunk 4 above)
Comment (bschoening) [reply pylib/cqlshlib/cqlshmain.py:2215]: Passing anything other than the current CQL version (3.4.6 w/4.1) causes an error:

$ cqlsh --cqlversion 3.4.2
....
Connection error: ('Unable to connect to any servers', {'127.0.0.1:9042': ProtocolError("cql_version '3.4.2' is not supported by remote (w/ native protocol). Supported versions: ['3.4.6']")})

File: pylib/cqlshlib/cqlshmain.py (hunk 1 above)
Comment (bschoening) [reply pylib/cqlshlib/cqlshmain.py:293]: The documentation at [Cassandra Query Language](https://cassandra.apache.org/doc/latest/cassandra/cql/changes.html) describes a valid version as only major/minor/patch. 

> Versioning of the CQL language adheres to the [Semantic Versioning](http://semver.org/) guidelines. Versions take the form X.Y.Z where X, Y, and Z are integer values representing major, minor, and patch level respectively. There is no correlation between Cassandra release versions and the CQL language version.

**Comment by bschoening** [comment]
Regarding field_size_limit, it was being parsed if defined but not used by the copyutil.py.

**Comment by ekaterinadimitrova2** [comment]
`Regarding field_size_limit, it was being parsed if defined but not used by the copyutil.py.`
The question is, why? Is it a bug or a leftover? I think we will need to do some archeology here

**Comment by bschoening** [comment]
field_size_limit seems non-functional since at least 2.1, it may have been used in 2.0 before the csv import/export was reworked.

**Comment by ekaterinadimitrova2** [comment]
@bschoening , @driftx - There is no test for this reported issue but  field_size_limit  was
 introduced in - https://issues.apache.org/jira/browse/CASSANDRA-8934 as a bug fix. It seems it was always "unused"? What do I miss? Can you take a look, please? After that one is confirmed, please feel free to commit this one. I will be off until Tuesday, no need to wait for me.

**Comment by bschoening** [comment]
As noted in the comments on the Jira, the implementation of field_size_limit is left over from Cassandra v2.0 before the copy from/copy to behavior was refactored into the copyutil.py file (2.1.13) in [CASSANDRA-9304](https://issues.apache.org/jira/browse/CASSANDRA-9304).  Copy options, not command line arguments, are now used to provide similar support for options. The field_size_limit arg to CQLSH its  been a non-functional since that time.
